edge/edge.go

```diff
@@ -92,6 +91,13 @@ func (e *edge) Stop() error {
 		return fmt.Errorf("edge: failed to stop message router: %w", err)
 	}
 
+	for _, service := range e.services {
+		err = service.Stop()
+		if err != nil {
+			return fmt.Errorf("edge: failed to stop service: %w", err)
```

Comment:
Do we really want to return the error if a service can't be stopped? What happens to the next services in the array? Can they be in an undesired state?